start/lib/Service/FleetLoader.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Service;
+
+use Model\Fleet;
+use Model\ShipFleet;
+
+
+class FleetLoader
+{
+    private $fleetStorage;
+    private $shipLoader;
+
+    public function __construct(FleetStorageInterface $fleetStorage, ShipLoader $shipLoader)
+    {
+        $this->fleetStorage = $fleetStorage;
+        $this->shipLoader = $shipLoader;
+    }
+
+    public function getFleetsByTeam(): array
+    {
+        $types = $this->fleetStorage->findTeams();
+        $fleetShipData = $this->fleetStorage->fetchFleets();
+
+        $fleets = [];
+        foreach ($types as $type) {
+            $fleets[$type] = [];
+        }
+
+        foreach ($fleetShipData as $fleetData) {
+            $fleet = $this->createFleetFromData($fleetData);
+            $fleetShip = $this->createFleetShipFromData($fleet, $fleetData);
+            $fleets[$fleet->getTeam()][] = $fleetShip;
+        }
+
+        return $fleets;
+    }
+
+    public function getFleetById($id): Fleet
+    {
+        $fleetData = $this->fleetStorage->findFleetById($id);
+
+        $fleet = $this->createFleetFromData($fleetData);
+
+        $fleet->setShipFleets($this->getFleetShipsByFleet($fleet));
+
+        return $fleet;
+    }
+
+    public function getFleetShipsByFleet(Fleet $fleet): array
+    {
+        $fleetShipsData = $this->fleetStorage->findFleetShipsByFleet($fleet);
+
+        $fleetShips = [];
+        foreach ($fleetShipsData as $fleetShipData) {
+            $fleetShips[] = $this->createFleetShipFromData($fleet, $fleetShipData);
+        }
+
+        return $fleetShips;
+    }
+
+    public function getFleetShipByIds($shipId, $fleetId): ShipFleet
```

Comment:
I think you may be able to remove this completely by adding a helper function in Fleet to do this same thing. I discuss this further in edit.php